daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala

```diff
@@ -276,9 +276,8 @@ private[daml] class EncodeV1(minor: LV.Minor) {
       */
     private val builtinFunctionInfos =
       DecodeV1.builtinFunctionInfos
-        .filterNot(info =>
-          versionIsOlderThan(info.minVersion) &&
-            info.maxVersion.forall(v => !versionIsOlderThan(v))
+        .filter(info =>
+          info.minVersion <= languageVersion && info.maxVersion.forall(languageVersion < _)
```

Comment:
This was wrong before right? At least it doesn’t seem equivalent to the old version.

Reply:
this should be the first fix. 
I fact  I am quite unable to decipher the original code.